packages/experimental/Avatar/src/Avatar.tsx

```diff
@@ -0,0 +1,176 @@
+
+const avatarName = 'MSFAvatarView';
+
+const NativeAvatarView = ensureNativeComponent(avatarName);
+
+export type Size = 'xSmall' | 'small' | 'medium' | 'large' | 'xLarge' | 'xxLarge';
+
+export type AvatarStyle = 'circle' | 'square';
+
+export type Presence = 'none' | 'available' | 'away' | 'doNotDisturb' | 'outOfOffice' | 'offline' | 'unknown' | 'unknown' | 'blocked';
+
+interface ExportedConstants {
+  sizes: { [key in Size]: number };
+}
+
+const ExportedNativeConstants: ExportedConstants = NativeModules.MSFAvatarViewManager;
+
+export type AvatarData = {
+  /**
+   * The primary text to create initials with (e.g. a name)
+   */
+  primaryText?: string;
+
+  /**
+   * The secondary text to create initials with if primary text is not provided (e.g. an email address)
+   */
+  secondaryText?: string;
+
+  /**
+   * The image to be displayed
+   */
+  image?: Image;
+
+  /**
+   * The color that represents this avatar.
+   * This color will override the initials view's background color.
+   * If the avatar view is configured to display a border, this will be the border's color.
+   * The colored border will not be displayed if a custom border image is provided.
+   */
+  color?: string;
+
+  /**
+   * Image to be used as border around the avatar. It will be used as a pattern image color,
+   * but It will be scaled to fit the avatar size. If set, the hasBorder initializer value will be ignored,
+   * since it's assumed that the client intends to have a custom border.
+   */
+  customBorderImage?: Image;
+
+  /**
+   * The avatar view's presence state.
+   * The presence state is only shown when the style is set to AvatarStyle.circle.
+   */
+  presence?: Presence;
+};
+
+export type otherAvatarProps = {
+  /**
+   * Supported Avatar sizes
+   */
+  size?: Size;
+
+  /**
+   * Background Color of initials
+   */
+  backgroundColor?: string;
+
+  /**
+   * Image to be used as border around the avatar. It will be used as a pattern image color,
+   * but It will be scaled to fit the avatar size. If set, the hasBorder initializer value will be ignored,
+   * since it's assumed that the client intends to have a custom border.
+   */
+  customBorderImage?: Image;
+
+  /**
+   * Shape of the AvatarView
+   * Circle is used to represent people
+   * Square is used to represent organizations or teams
+   */
+  avatarStyle?: AvatarStyle;
+
+  /**
+   * The color of the border of the avatar view
+   */
+  borderColor?: string;
+
+  /**
+   * When true, the presence status border is opaque. Otherwise, it is transparent.
+   */
+  usesOpaquePresenceBorder?: boolean;
+
+  // TODO Map the viewProp AccessibilityLabel to this
+  // overrideAccessibilityLabel? : string;
+
+  /**
+   * Used when avatarView doesn't have image or can't generate initials string
+   */
+  preferredFallBackImageStyle?: string;
+
+  /**
+   * Set to true to enable the pointer interaction on the avatar view, false by default.
+   * TODO Figure out how to make iOS only
```

Comment:
For now commenting the platforms it supports is a good start. Maybe @JasonVMo knows how we've been doing platform specific props?

Reply:
I think there are a few ways to think about props. Beyond the props that just apply to all platforms, the next preference would be something that has:

- A name and semantic meaning that isn't platform specific
- Implementations for platforms where it applies, no-op for ones where it doesn't
- Comments denoting the expected behaviors and where it applies

If on the other hand it is truly platform specific, then the pattern seems to be to either put the platform in the prop name or fork the interface. Given the fact that TypeScript doesn't work well with interfaces that change shape per platform I'd prefer either semantic names or platform specific names to forked interfaces.

Reply:
This one seems to fall in the first bucket, enabling pointer interaction seems like a switch that could apply to any platform.